File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    let newArr = []
    for (let i = 2; i <= n; i++) {
        newArr.push(i)
    }
    
    let prime = []
    for (let i = 2; i <= n; i++) {
        if (newArr[i] === 0) continue
        prime.push(i)
        newArr[i] = 0
        
        for (let j = i*2; j <= n; j += i) {
            if (newArr[j] === 0) continue
            newArr[j] = 0
        }
    }
    return prime.length
}

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    is_prime = [True] * (n + 1)
    is_prime[0] = False
    is_prime[1] = False
    
    for i in range(2, int(n**0.5) + 1):
        if is_prime[i]:
            for j in range(i * i, n + 1, i):
                is_prime[j] = False
                
    count = 0
    for i in range(2, n + 1):
        if is_prime[i]:
            count += 1
    return count